test/dotnet-core-uninstall.Tests/Shared/Commands/CommandBundleFilterTests.cs

```diff
@@ -36,9 +36,9 @@ public class CommandBundleFilterTests
         };
 
         [WindowsOnlyTheory]
-        [InlineData("remove --all --sdk", new string[] { "1.0.0", "1.0.1" })]
-        [InlineData("dry-run --all --sdk", new string[] { "1.0.0", "1.0.1" })]
-        [InlineData("whatif --all --sdk", new string[] { "1.0.0", "1.0.1" })]
+        [InlineData("remove --all --sdk", new string[] { "1.0.0", "1.0.1", "7.0.1" })]
```

Comment:
Why was 7 needed to be added here? (I'm just comparing to our last TFM update and looking at other data inputs for this test method)

Reply:
To be totally honest, I have no recollection of anything having to do with this PR anymore, so I have no idea.

Reply:
ahh, so in the other file we need to add additional buckets?

Reply:
If we don't want 7.0.1 to be uninstallable, then yes. But then I have to ask why 7.0.1 shouldn't be uninstallable...and also why some of those other versions shouldn't be uninstallable. Are we really going to say we can't uninstall 1.0.0 unless we have 1.1.0? Or that we want to keep both 2.1.200 and 2.1.300? Most of those are out-of-support; part of me wonders if we can just delete everything before the 3.0.100 line. This is definitely much more your area, though, so if you think we should just add more buckets, I can do that.

Reply:
My understanding is that the top entry for each band is uninstallable because it could be used by VS or VSMac. I'd have to dig into the code to confirm that but the help indicates that.

The more I dig into these tests, the more I think the tests kind of wonky. I tried to improve them but it's a lot to unwind. I'll poke more at a simpler test fix.

@joeloff @baronfel thoughts?

Reply:
Ugh, the tool is making silly guesses. we haven't shipped bundles in VS since 16.2 and .NET Core 2.2. A standalone bundle doesn't imply that VS is using it, nor does the highest band because you can have an unlimited number of instances from 15.11 through 17.9 covering each feature band.

Reply:
Sounds like joeloff recommends deprecating the uninstall tool entirely since we can't get it right anyway 😜

Reply:
@joeloff are you saying the tool is silly or the tests are silly? I think I'm close to getting a simple fix for the tests which adding in the buckets we were talking about above and adds in new items in the version object. I am just not sure about some of the true/false values for some of the other tests. I'll push something in the morning.

Reply:
The tool is silly :). Back when this was designed, .NET used to ship the standalone SDK bundles in VS. Bundles in VS require overhead to track. For example, we shipped both 1.0 and 1.1 SDKs in VS 15 (eventually 2.0, 2.1, and 2.2). The tools is simply making an assumption that if you have something like the 1.1.6 SDK (which was the last version we shipped for 1.1) then it's likely that it's used by VS. And leaving the highest version of a major.minor .NET SDK on the machine is less likely to break someone.